### PR TITLE
Adding more cmd line options to run_scans.py

### PR DESCRIPTION
**Pull request by bdorney**
Addressing:

https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/107
https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/106
https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/112

@cbravo135 and @mexanick 

Additionally added a new dictionary in chamberInfo.py:

`chamber_vfatMask`

This will now allow run_scans.py to send vfatmask parameter correctly to each chamber.  The defaults assigned for P5 are 0x0 for all links, 0x0 for coffin setup, and 0xF40400 for the cosmic stand.

Now also requires:
https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/13

### CODE REVIEW

File: trimChamber.py [hunk 1]
```diff
@@ -86,8 +86,10 @@
             "-s%d"%(options.slot),
             "-g%d"%(options.gtx),
             "--filename=%s"%(filename0),
-	        "--vfatmask=%i"%(options.vfatmask)]
-           )
+            "--vfatmask=%i"%(options.vfatmask),
```
Comment (jsturdy) [inline trimChamber.py]: This is definitely the wrong way to do this, since there is no expectation that the `vfatmask` will be the same for all chambers under consideration, probably best would be to have it stored in the `chamber_info` file per chamber

Comment (bdorney) [reply trimChamber.py]: Actually trimChamber.py seems to be the place to put this, from what I understood of your comment though run_scans.py should read in a dict from chamber_info and give the vfatmask option to the chamber of interest when called as it does per the other values.

I will update run_scans.py to have this functionality.

Comment (bdorney) [reply trimChamber.py]: Functionality now included in chamber_vfatMask dictionary and used by run_scans.py

File: run_scans.py [hunk 2]
```diff
@@ -21,7 +22,7 @@ def launchTestsArgs(tool, shelf, slot, link, chamber, scanmin, scanmax, nevts,
   #Build Commands
   setupCmds = []
   preCmd = None
-  cmd = ["%s"%(tool),"-s%d"%(slot),"-g%d"%(link),"--shelf=%i"%(shelf)]
+  cmd = ["%s"%(tool),"-s%d"%(slot),"-g%d"%(link),"--shelf=%i"%(shelf), "--nevts=%i"%(nevts), "--vfatmask=%s"%(vfatmask)]
```
Comment (jsturdy) [inline run_scans.py]: you've tested this?  some places you're passing it as an `int` (`%i`), some places as a `string` (`%s`)
I'd have probably naively gone with `hex` (`%x` or `0x%x`)

Comment (bdorney) [reply run_scans.py]: Mmhmm...an oversight for sure.  I'll try to incorporate the suggesting and then re-test.

File: trimChamber.py [hunk 3]
```diff
@@ -232,7 +232,7 @@
             "-s%d"%(options.slot),
             "-g%d"%(options.gtx),
             "--filename=%s"%(filenameFinal),
-	        "--vfatmask=%i"%(options.vfatmask),
+	        "--vfatmask=%x"%(options.vfatmask),
```
Comment (jsturdy) [inline trimChamber.py]: check tabs here (L235)

File: run_scans.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def launchTestsArgs(tool, shelf, slot, link, chamber, vfatmask, scanmin, scanmax
   #Build Commands
   setupCmds = []
   preCmd = None
-  cmd = ["%s"%(tool),"-s%d"%(slot),"-g%d"%(link),"--shelf=%i"%(shelf), "--nevts=%i"%(nevts), "--vfatmask=%s"%(vfatmask)]
+  cmd = ["%s"%(tool),"-s%i"%(slot),"-g%i"%(link),"--shelf=%i"%(shelf), "--nevts=%i"%(nevts), "--vfatmask=%i"%(vfatmask)]
```
Comment (jsturdy) [inline run_scans.py]: probably doesn't affect operation, but for consistency, change L25 to `--vfatmask=0x%x` (unless while testing this was found to not work)

File: qcoptions.py [hunk 5]
```diff
@@ -8,7 +8,7 @@
                   help="Minimum value of scan parameter", metavar="scanmin", default=0)
 parser.add_option("--scanmax", type="int", dest="scanmax",
                   help="Maximum value of scan parameter", metavar="scanmax", default=254)
-parser.add_option("--vfatmask", type="int", dest="vfatmask",
+parser.add_option("--vfatmask", type="int", dest="vfatmask",
```
Comment (jsturdy) [inline qcoptions.py]: And since I'm making other change requests, prefer to remove all trailing whitespace (L11)

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: run_scans.py (hunk 4 above)
Comment (bdorney) [reply run_scans.py]: I had tried `--vfatmask=%x` and it didn't work (learned why it should have been `--vfatmask=0x%x`) after googling more about optparser class:

https://docs.python.org/2/library/optparse.html#optparse-standard-option-types

Relying on optparser to identify the type here also allows the user to enter both hex or a binary number (whereas the `--vfatmask=0x%x` does not) which did work in testing.  Since optparser automatically handles this I don't think the change is necessary.

File: trimChamber.py (hunk 3 above)
Comment (bdorney) [reply trimChamber.py]: Mmhmm, not sure how that slipped through

File: run_scans.py (hunk 4 above)
Comment (jsturdy) [reply run_scans.py]: OK, but the point here was that the user is not going to be specifying anything in this code (the value is passed in from somewhere else), so what is printed to the screen and logfile, should be as readable as possible (decimal representation is the least readable and binary is long).

Comment (bdorney) [reply run_scans.py]: Okay, understood.  I will change to:

`"--vfatmask=0x%x"`

And make the appropriate modifications in chamberInfo.py

But in this case the log file will *still* show the number as decimal in some instances since optparser itself does the conversion. e.g. using the `--debug` option with `run_scans.py` will have a hex argument:

0xF40400

Displayed as:

15991808

Due to the conversion by optparser.

File: qcoptions.py (hunk 5 above)
Comment (jsturdy) [reply qcoptions.py]: Still trailing whitespace (since github told me my comment was outdated)

File: run_scans.py (hunk 4 above)
Comment (jsturdy) [reply run_scans.py]: Which line makes this print out?  We should just fix that.
Also, what needs to change in `chamberInfo.py`?  The version in this PR looked fine just a minute ago

File: qcoptions.py (hunk 5 above)
Comment (bdorney) [reply qcoptions.py]: Mmhmm, maybe It's something in my .vimrc then...from terminal in my editor I do not see these extra spaces but they do appear:

https://github.com/bdorney/vfatqc-python-scripts/blob/master/qcoptions.py

If I highlight the text in github...

File: run_scans.py (hunk 4 above)
Comment (bdorney) [reply run_scans.py]: Looks like:

https://github.com/bdorney/vfatqc-python-scripts/blob/master/run_scans.py#L198

Maybe calling `hex()` here will ensure the log file has this written in hex format.

For chamberInfo.py if we have:

`--vfatmask=0x%x`

I understood from:

https://docs.python.org/2/library/optparse.html#optparse-standard-option-types

That the number should be formatted like "0x...." but from [chamber_vfatmask](https://github.com/bdorney/vfatqc-python-scripts/blob/bc120ab04e6b20eb302585deeaf6d2bda0bd4ddf/chamberInfo.py#L37)

The inputs are `0x0` so they should instead be stored as `0` otherwise we would pass `0x0x0` to vfatmask which in the `append()` command is a string so it is not interpreted as a hex number.

Basically we need to resolve 0x... vs. 0x0x... or have I misunderstood?

Comment (jsturdy) [reply run_scans.py]: You've (possibly) misunderstood, once the number is extracted from the `chamberInfo.py` it is an `int`, so it's machine readable, then when you pass this to the line we're discussing, it will be correctly converted to `hex`.
Though I'd double check with some test, just to convince yourself

File: qcoptions.py (hunk 5 above)
Comment (jsturdy) [reply qcoptions.py]: could be the way that github manages comments linked to code, as i had a similar issue just now where what i see on the PR dashboard and what I see in the latest version of the file in the PR aren't in agreement...
UPDATE nope, just checked and confirm what you've seen, could be that it's a single space, or a tab (not sure what `vim` is showing you?

Comment (bdorney) [reply qcoptions.py]: I added some statements to my ~/.vimrc to remove trailing white space on *.py files whenever the save action is performed by vim (e.g. `:w`).  Hopefully this should solve the problem...but git may not have realized anything in the file has changed.

**Review by jsturdy — APPROVED** [review]
ok, if it works, i sign off

**Comment by bdorney** [comment]
Unit tests completed, tests on cosmic stand using `run_scans.py` successful:

http://cmsonline.cern.ch/cms-elog/1003922

Will merge once cms-gem-daq-project/gem-plotting-tools#13 is approved (since this PR is dependent)

**Comment by bdorney** [comment]
https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/13

Has been merged.  Now merging this request
